crates/grafbase-sdk/src/types.rs

```diff
@@ -0,0 +1,113 @@
+//! Type definitions of the input and output data structures of the SDK.
+
+pub use minicbor_serde::error::DecodeError;
+pub use serde::Deserialize;
+use serde::Serialize;
+
+/// The directive and its arguments which define the extension in the GraphQL SDK.
+pub struct Directive(crate::wit::Directive);
+
+impl Directive {
+    /// The name of the directive.
+    pub fn name(&self) -> &str {
+        &self.0.name
+    }
+
+    /// The directive arguments. The output is a Serde structure, that must map to
+    /// the arguments of the directive.
+    ///
+    /// Error is returned if the directive argument does not match the output structure.
+    pub fn args<'de, T>(&'de self) -> Result<T, DecodeError>
```

Comment:
We shouldn't expose `DecodeError` as part of our API. Should be `impl Error` for example. Would also just `arguments` instead of `args` IMHO.

Reply:
It's a bit annoying with `impl Error` because the user needs to define the error type. Maybe we should have our own Error enum with thiserror here?